OptimoveSDK/optimove-sdk/src/main/java/com/optimove/android/preferencecenter/OptimovePreferenceCenter.java

```diff
@@ -0,0 +1,300 @@
+        switch (region) {
+            case "uk-1":
+                return "dev-pb";
+            case "eu-central-2":
+                return "eu";
+            case "us-east-1":
+                return "us";
+            default:
+                throw new IllegalArgumentException("Region is not supported: " + region);
+        }
+    }
+
+    private static JSONArray mapPreferenceUpdatesToArray(List<PreferenceUpdate> updates) throws JSONException {
+        List<Object> mappedUpdates = new ArrayList<>();
+
+        for (int i = 0; i < updates.size(); i++) {
+            String updateId = updates.get(i).getTopicId();
+            List<Channel> channels = updates.get(i).getSubscribedChannels();
+            List<Integer> mappedChannels = new ArrayList<>();
+            for (int j = 0; j < channels.size(); j++) {
+                mappedChannels.add(channels.get(i).getValue());
+            }
+
+            Object mappedUpdate = new Object() {
+                final String id = updateId;
+                final List<Integer> subscribedChannels = mappedChannels;
+            };
+            mappedUpdates.add(mappedUpdate);
+        }
+
+        return new JSONArray(mappedUpdates.toArray());
+    }
+
+    private static Preferences mapResponseToPreferences(Response response) {
+        try {
+            JSONObject data = new JSONObject(response.body().string());
+
+            JSONArray channels = data.getJSONArray("channels");
+            List<Channel> configuredChannels = new ArrayList<>();
+
+            int len = channels.length();
+            for (int i = 0; i < len; i++) {
+                configuredChannels.add(Channel.getChannelByValue(channels.getInt(i)));
+            }
+
+            JSONArray topics = data.getJSONArray("topics");
+            List<Topic> customerPreferences = new ArrayList<>();
+
+            int topicLength = topics.length();
+            for (int i = 0; i < topicLength; i++) {
+                JSONObject topicObj = topics.getJSONObject(i);
+
+                List<Channel> subscribedChannels = new ArrayList<>();
+                JSONArray channelSubscriptionArray = topicObj.getJSONArray("channelSubscription");
+                for (int j = 0; j < channelSubscriptionArray.length(); j++) {
+                    subscribedChannels.add(Channel.getChannelByValue(channelSubscriptionArray.getInt(j)));
+                }
+
+                Topic topic = new Topic(topicObj.getString("topicId"), topicObj.getString("topicName"), topicObj.getString("topicDescription"), subscribedChannels);
+
+                customerPreferences.add(topic);
+            }
+
+            return new Preferences(configuredChannels, customerPreferences);
+        } catch (NullPointerException | JSONException | IOException e) {
+            Log.d(TAG, e.getMessage());
+            return null;
+        }
+    }
+
+    private static void logFailedResponse(Response response) {
+        switch (response.code()) {
+            case 404:
```

Comment:
are these status codes actually possible? 

1. if you give random customer id -- it actually returns 200 with everything enabled. Reason being that by default everything is enabled for a customer (mb bit weird, but that's how it works)
2. if you give bad tenantId/brand group id -- it gives 400